docs/GAS.md

```diff
@@ -0,0 +1,40 @@
+# Gas
+
+Gas is a way to measure computational expense of a smart contract execution,
+including CPU time and storage cost. It's unit is 1, i.e. you can think of it as
+countable points. Gas consumption is deterministic, so executing the same thing
+costs the same amount of gas across all hardware and operating systems.
+
+CosmWasm charges gas for Wasm operations, calls to host functions and calls to
+the Cosmos SDK. CosmWasm gas is different from Cosmos SDK gas as the numbers
+here are much larger. Since we charge gas for arbitrary user defined operations,
+we need to charge each Wasm operation individually and cannot group larger tasks
+together. As a result, the gas values become much larger than in Cosmos SDK even
+for very fast executions. There is a [multiplier][defaultgasmultiplier] to
+translate between CosmWasm gas and Cosmos SDK. It was measured and set to 100 a
+while ago and can be adjusted when necessary.
+
+For CosmWasm gas, the target gas consumption is 1 Terragas (10^12 gas) per
+millisecond. This idea is [inspired by NEAR][neargas] and we encourage you to
+read their excellent docs on that topic.
+
+In order to meet this target, we execute Argon2 in a test contract ([#1120]).
```

Comment:
This is a great document.

The one question I raise is the reproduceability of "this consumed 96837752 gas and took 15ms on our CI system". They may upgrade the CI sometime.

It is fine for now, but we should have a better benchmark system sometime. (Can be in a few months)